src/Symfony/Component/Scheduler/Scheduler.php

```diff
@@ -62,9 +66,17 @@ public function run(array $options = []): void
 
             $ran = false;
             foreach ($this->generators as $generator) {
-                foreach ($generator->getMessages() as $message) {
-                    $this->handlers[$message::class]($message);
+                foreach ($generator->getMessages() as $context => $message) {
+                    $preRunEvent = new PreRunEvent($generator->schedule(), $context, $message);
```

Comment:
Nit: maybe could we avoid creating this event if `$this->dispatcher` is null?

Reply:
I will update!